tools/topology/topology1/sof-adl-nau8825.m4

```diff
@@ -0,0 +1,350 @@
+#
+# Topology for Alderlake with headphone, DMIC, HDMI with/without speaker.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+include(`hda.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Alderlake DSP configuration
+include(`platform/intel/adl.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+ifdef(`AMP_SSP',`ifdef(`SMART_AMP', , `
+include(`muxdemux.m4')
+
+#
+# Define the demux configure
+#
+# PCM0 --> volume --> demux --> SSP1/SSP2 (speaker - max98360a/rt1019)
+#                       |
+# PCM6 <----------------+
+
+ifdef(`AMP_SSP',`',`fatal_error(note: Define AMP_SSP for speaker amp SSP Index)')
```

Comment:
see here, AMP_SSP is required, so there's no point in using the ifdef(`AMP_SSP',`ifdef(`SMART_AMP', ,  nested test

Reply:
@plbossart Yes, you are right! I will remove the redundant ifdef below:
"ifdef(`AMP_SSP',`',`fatal_error(note: Define AMP_SSP for speaker amp SSP Index)')"